index.js

```diff
@@ -0,0 +1,66 @@
+
+function checkSalaryTotal(lineup) {
+
+  return (45000 > lineup.map((lineup) => lineup.salary).reduce((total, salary) => total + salary, 0))
+}
+
+function checkTeams(lineup) {
+  const teams = lineup.map((lineup) => lineup.teamId)
+  const sortedTeams = teams.sort((a, b) => a - b)
+  let currentCount = 1
+  let highestSteak = 1
+
+  for (let i = 0; i < sortedTeams.length; i++) {
+    if (sortedTeams[i] === sortedTeams[i + 1]) {
+      currentCount++
+    } else {
+      currentCount = 1
+    }
+    if (currentCount > highestSteak) {
+      highestSteak = currentCount
+    }
+  }
+
+  return highestSteak <= 2
+}
+
+function checkGames(lineup) {
+  const games = lineup.map((lineup) => lineup.gameId)
+  const sortedGames = games.sort((a, b) => a - b)
+  let currentCount = 1
+  let highestSteak = 1
+
+  for (let i = 0; i < sortedGames.length; i++) {
+    if (sortedGames[i] === sortedGames[i + 1]) {
+      currentCount++
+    } else {
+      currentCount = 1
+    }
+    if (currentCount > highestSteak) {
+      highestSteak = currentCount
+    }
+  }
+
+  return highestSteak <= 3
+}
+
+function checkPositions(lineup) {
+  const properPositionLineup = ['C', 'OF', 'OF', 'OF', 'P', 'SS', '1B', '2B', '3B']
+  const positions = lineup.map((lineup) => lineup.position)
+
+  return positions.every(element => properPositionLineup.indexOf(element) > -1)
+}
```

Comment:
This doesn't actually check the positions correctly. It makes sure that the positions seen are in the list but it doesn't ensure that the right number of positions are present.

![image](https://user-images.githubusercontent.com/5775716/78806518-c5dbf680-7990-11ea-9e08-e4a16f39238e.png)